src/cli/cli.cpp

```diff
@@ -4203,6 +4247,24 @@ template <> otError Interpreter::Process<Cmd("mode")>(Arg aArgs[])
         ExitNow();
     }
 
+    /**
+     * @cli mode (get,set)
```

Comment:
This block should go before line 4235 above 

`template <> otError Interpreter::Process<Cmd("mode")>(Arg aArgs[])`

Also we should link to both the get & set APIs, so we'll need to include text as `@par` tags and use `@sa` for that.  See https://github.com/openthread/openthread/blob/b2116446bc79230a7afec49fb421d538bbb6fbdc/src/cli/cli_bbr.cpp#L285 for an example.